core/src/main/scala/org/http4s/headers/`Idempotency-Key`.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.headers
+
+import org.http4s.internal.parsing.Rfc7230.{quotedString, token}
+import org.http4s.{Header, ParseResult}
+import org.typelevel.ci._
+
+object `Idempotency-Key` {
+  def parse(s: String): ParseResult[`Idempotency-Key`] =
+    ParseResult.fromParser(parser, "Invalid Idempotency-Key header")(s)
+
+  private[http4s] val parser =
+    token.orElse(quotedString).map(CIString(_)).map(`Idempotency-Key`.apply)
+
+  implicit val headerInstance: Header[`Idempotency-Key`, Header.Single] =
+    Header.createRendered(
+      ci"Idempotency-Key",
+      _.key,
+      parse
+    )
+}
+
+/** Request header defines request to be idempotent used by client retry middleware.
+  *
+  *  [[https://tools.ietf.org/html/draft-idempotency-header-00#section-2.1 idempotency-header]]
+  */
+final case class `Idempotency-Key`(key: CIString)
```

Comment:
Oh, one more thing: is there anything in the spec that suggests this key is case-insensitive?  It seems to me that it's just an opaque token.